### ISSUE TITLE
Global padding not applied in the template part editor

### ISSUE THREAD

**Issue by lunule** [WordPress/gutenberg#59742]
### Description

The global padding feature, when `useRootPaddingAwareAlignments` is set to `true` in the `theme.json` file, is not working properly in the site editor: 
- while global paddings are applied in the _template editor_,
- the same global paddings are not applied in the _template part editor_ - the `has-global-padding` class is applied correctly, but the class styles are missing.

**Template editor** => _header with horizontal paddings displayed **correctly**_

![global-padding--template-editor](https://github.com/WordPress/gutenberg/assets/190486/c63d1be2-d756-4881-8542-0ba482cb53ee)

**Template part editor** => _header with horizontal paddings displayed **incorrectly**_

![global-padding--template-part-editor](https://github.com/WordPress/gutenberg/assets/190486/694699aa-da03-4174-8c8f-0c91a09d785b)

### Step-by-step reproduction instructions

1. Use a `layout.type = constrained` div (directly following a `layout.type = flow` div, or one without `layout.type` specification) _in a template part_
2. Set up non-zero `styles.spacing.padding` values in the `theme.json` file
3. Test the padding display 
   - in the template editor at first
   - then in the template part editor

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
I agree, that the padding is missing is confusing to me too when I am working with the template parts in "isolation".
To me, it makes it more difficult to create designs. Because I keep thinking that I need to add the spacing to the actual content inside the part. If the user does that, the padding is of course duplicated when you view the full page or template.
Also, the global styles panel is available and it is possible to adjust the site wide spacing from this screen, in the Styles > Layout panel, event hough it is not applied and you can't see it.

I was told that this is intentional. I'm not sure if I can find the conversation, if I do stumble upon it again I will add a link here.

**Comment by lunule** [comment]
> To me, it makes it more difficult to create designs. Because I keep thinking that I need to add the spacing to the actual content inside the part.

I agree - this is the exact reason why I think this current implementation is not developer-, designer-, or user-friendly. It's a deceptive experince while working on theme design.

> I was told that this is intentional. I'm not sure if I can find the conversation, if I do stumble upon it again I will add a link here.

Thanks - I would think the reasoning is about that the template part editor should display content and styles used strictly in the part design, while global padding is, well, _global_, so it should be displayed in the larger context only, which is the template, not some of its parts (_even if the global padding is practically applied only on a container element of a template part_).

Personally, I would partially agree with such reasoning, but overall, I would also vote against it because this is a _theory vs. practice/experience_ type of topic, and, as the purpose of the whole Full Site Editing paradigm is providing site owners with an easy-to-use and easy-to-learn site editor and site _editing experience_ - and this easy-to-misinterpret experience definitely goes against the principle.

**Comment by annezazu** [comment]
Hey folks! Here's the discussion if I have it correct: https://github.com/WordPress/gutenberg/issues/51937 I've removed the bug label for now.

**Comment by iamtakashi** [comment]
I agree. The discrepancy in appearance between the site editor and the template part editor doesn't make sense to me, even though it's technically correct. I assume it doesn't make sense to the vast majority of users too. In my opinion, it should appear identically in both editors.
